### DISCUSSION TITLE
using `gum table` or `gum pager` as pager for csv data

### DISCUSSION THREAD

**Discussion by gennaro-tedesco** [charmbracelet/gum#458] [General]
I am trying to use either of the two options `table` or `pager` as terminal pager to display tabular data (or to pipe the results of tabular data into them): however, I have the impression that both of them come a little short.

For instance:

### gum table
- would it be possible to search within the table (similar to what one does with `/` in `less` or others)? I have not been able to find out how to do so (and at the moment this is limiting me from using it fully)
- what is the right grammar to expand width of all columns at once (instead of using `-w X,Y,Z...`, which assumes you know already the shape of the data you are displaying)? Often we do not know in advance how many/which columns we want to expand (even here I presume such option is there, I could not find it in the docs?)

### gum pager
equivalently, the pager could be used instead of the table, however
- is it possible to fix the headers for scrolling (similar to what happens in `less --header 1`, namely having sticky headers)?
- is it possible to scroll horizontally and toggle wrap/unwrap for columns? In many cases a lot of data are cut off due to terminal width, and the pager is unable to display them correctly

How are you using the two options above (if at all) to display tabular data?

**Comment by MikaelFangel** [comment]
Hi,
It would certainly be possible to write a search functionality for `gum table` like there is for `gum pager`. I'm the author of the pager one. However, one question comes to mind for it to be really useful, should it be a fully fledge search functionality like the one on the pager or would a fuzzy matcher only showing lines with matches be fine? Like `gum filter` but for tables.

 * I don't see why writing something that the mimics `less --header 1` would be an issue, but again, implementation choices I think should be made by the charmbracelet team.
 * In `gum pager` currently, only line wrapping is supported, and side scrolling is on the wish list.

And maybe, while this is not exactly what you're looking for, you can get a somewhat more readable table in the gum pager by doing the following:

```bash
cat my.csv | sed $'s/,/\t/g' | gum pager --soft-wrap
```

**Comment by gennaro-tedesco** [comment]
>  should it be a fully fledge search functionality like the one on the pager or would a fuzzy matcher only showing lines with matches be fine?

Ideally I presume users would expect pagers to have the same behaviour as `less`, namely to be able to jump from (highlighted) match to match with `n/N`; obviously if too complex one may deliver this in steps and start with a row reference first.

I agree that implementation details ought to be agree by the charmbracelet team, hopefully they jump in with some ideas and/or suggestions.

**Reply by MikaelFangel** [reply]
Hm, I don't think it is too complex to implement. My main concern is if the searcher from gum pager can be generalised so it can be used by both. So it's more maintainable in the future.

**Reply by gennaro-tedesco** [reply]
You're a making a good point, it would be most practical to write one searcher only that could ideally be "plugged" into any of the `gum` commands. Perhaps here is where input from the maintainers is needed, they would know best if such choice is preferable.
